lib/template/tests/test_template_on_error.c

```diff
@@ -25,17 +25,17 @@
 #include <libgen.h>
 
 #include "template/templates.h"
-#include "testutils.h"
+#include <criterion/criterion.h>
 #include "apphook.h"
 
 #define assert_on_error_parse(on_error,expected)                        \
```

Comment:
You can eliminate these macros due to criterion:
For now we a single test with five asserts, four for success case, one for failure. This single test should be split into two tests: a parametrized criterion test for the four success cases, and one regular test for the failure case.

About parameterezed test, I mean those that are defined with `ParameterizedTest` and `ParameterizedTestParameters`.

Reply:
Hi @furiel ,

I am exploring ParameterizedTest and will make changes as you suggested.

Reply:
Thank you! The test looks beautiful. :)